core/chainio/avs_writer.go

```diff
@@ -101,14 +112,65 @@ func (w *AvsWriter) SendAggregatedResponse(ctx context.Context, batchMerkleRoot
 		return nil, err
 	}
 
-	receipt, err := utils.WaitForTransactionReceipt(w.Client, ctx, tx.Hash())
+	txNonce := big.NewInt(int64(tx.Nonce()))
+	receipt, err := w.WaitForTransactionReceiptWithIncreasingTip(ctx, tx.Hash(), txNonce, batchMerkleRoot, nonSignerStakesAndSignature)
+	w.logger.Info("Transaction receipt:", "receipt", receipt)
 	if err != nil {
 		return nil, err
 	}
 
 	return receipt, nil
 }
 
+func (w *AvsWriter) WaitForTransactionReceiptWithIncreasingTip(ctx context.Context, txHash gethcommon.Hash, txNonce *big.Int, batchMerkleRoot [32]byte, nonSignerStakesAndSignature servicemanager.IBLSSignatureCheckerNonSignerStakesAndSignature) (*gethtypes.Receipt, error) {
+	currentSleepTime := 0 * time.Second
```

Comment:
I think logic for this function should be changed a bit, maybe the retries should be actually retries increasing the tip, and the wait for tx receipt can be using the old function with some config params. The loop would be:
1. WaitForTxReceipt (e.g 5 retries every 4 seconds)
    a. If receipt is found return everything ok 
2. Increase Tip Cap and resend tx 

I think this is the current logic, but its not very clear with the %

Reply:
Solved in [ebfb204](https://github.com/yetanotherco/aligned_layer/pull/330/commits/ebfb204fc7badbf4e955be465d27e04d9ba4a988).